src/main/java/com/commercetools/sync/sdk2/types/utils/FieldDefinitionUpdateActionUtils.java

```diff
@@ -200,6 +203,10 @@ static Optional<TypeUpdateAction> buildChangeInputHintUpdateAction(
       @Nonnull final FieldDefinition oldFieldDefinition,
       @Nonnull final FieldDefinition newFieldDefinition) {
 
+    if (newFieldDefinition.getInputHint() == null && oldFieldDefinition.getInputHint() != null) {
+      return Optional.empty();
```

Comment:
I think it's better to throw an `UnsupportedOperationException` here?

Reply:
Interesting thing: for product type we also have inputHint and [there the default value is SINGLE_LINE](https://github.com/commercetools/commercetools-sync-java/blob/6281d8061da5ae924a2e2247bbe988fd0895518f/src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionUpdateActionUtils.java#L272), even though [the docs](https://docs.commercetools.com/api/projects/productTypes#change-attributedefinition-inputhint) do not say anything about it. Can you check? Try to do the update action on the type without `inputHint` to see if it would pass and the same for product type.

Reply:
The SDK does throw a NPE when `inputHint` / `newValue` isn't provided in the builders. When passing `null` to the API, the platform throws BadRequestException in both cases, because it's a required field. 
In both, ProductType and Type API the `inputHint` has a default value "SingleLine", set by CTP.

The [docs](https://docs.commercetools.com/api/projects/types#fielddefinition) for FieldDefinitions say the default value is SINGLE_LINE for inputHint, so i've changed the implementation. [d26a791](https://github.com/commercetools/commercetools-sync-java/pull/1013/commits/d26a791913ce15c40a37aacaa7d1feebfd72cccd)